### ISSUE TITLE
Incorrectly flags missing arguments for `attrs` fields with leading underscores

### ISSUE THREAD

**Issue by sasirven** [microsoft/pyright#11082]
## Issue summary:

Pyright incorrectly reports missing arguments when using `attrs` with fields that have leading underscores, even though `attrs` strips leading underscores from argument names by default unless an alias is explicitly set.

## Code to reproduce:

```python
import attrs

@attrs.define
class Test:
    _x: int
    _y: int

t = Test(x=1, y=2)  # Pyright error: Expected arguments '_x' and '_y'
```

## Expected behavior:

This should work without errors. According to `attrs` [documentation](https://www.attrs.org/en/stable/api.html), the generated `__init__` method strips leading underscores from argument names unless an alias is provided. So `Test(x=1, y=2)` is valid.

## Workaround:

To silence Pyright, I have to explicitly set aliases:

```python
@attrs.define
class Test:
    _x: int = attrs.field(alias="x")
    _y: int = attrs.field(alias="y")

t = Test(x=1, y=2)  # No Pyright error
```
## Why this is a bug:

Pyright does not respect `attrs`' default behavior of stripping leading underscores from argument names. This leads to false-positive errors and forces unnecessary boilerplate code.

## Environment:

Pyright version: 3.14
Python version: 1.1.407
attrs version: 25.4.0

**Comment by erictraut** [comment]
Pyright is working as designed here, so I don't consider this a bug.

Pyright has no knowledge of `attrs`-specific behaviors. It bases its analysis on standard type checking rules as specified in the Python typing spec. The `attrs` library uses a feature of the typing system called [dataclass_transform](https://typing.python.org/en/latest/spec/dataclasses.html#the-dataclass-transform-decorator) to indicate that it has dataclass-like semantics. The `attrs` library deviates from these semantics in some cases, such as how it handles attributes that start with an underscore. If you want this to type check, you'll need to modify your code or otherwise work around attr's behavior in this case.

**Comment by sasirven** [comment]
Okay, thanks for the quick response and clear explanation! I'll keep using 'alias' in that case.
